GroundDBMemoryPool/src/rdma/util.cc

```diff
@@ -285,19 +285,19 @@ int post_send(const struct resources *res, const struct memory_region *memreg, c
  * Description
  *
  ******************************************************************************/
-int post_receive(const struct resources *res, const struct memory_region *memregs, const struct connection *conn){
-    return post_receive(memregs, conn);
+int post_receive(const struct resources *res, const struct memory_region *memregs, const struct connection *conn, size_t lofs, size_t size){
+    return post_receive(memregs, conn, lofs, size);
 }
-int post_receive(const struct memory_region *memreg, const struct connection *conn)
+int post_receive(const struct memory_region *memreg, const struct connection *conn, size_t lofs, size_t size)
 {
     struct ibv_recv_wr rr;
     struct ibv_sge sge;
     struct ibv_recv_wr *bad_wr;
     int rc;
     /* prepare the scatter/gather entry */
     memset(&sge, 0, sizeof(sge));
-    sge.addr = (uintptr_t)memreg->buf;
-    sge.length = BUFFER_SIZE;
+    sge.addr = (uintptr_t)memreg->buf + lofs;
```

Comment:
Cast to char*?

Reply:
How does `char*` differ from `uintptr_t`?

Reply:
It's all right. I mistaken pointer arithmetic.